python/tank/platform/engine.py

```diff
@@ -910,7 +910,7 @@ def register_command(self, name, callback, properties=None):
 
         - ``group_default`` - Boolean value indicating whether this command should represent the group
           as a whole. Setting this value to True indicates that this is the command to run and display
-          in applications that show a single button for each command group.
+          in applications that show a single representation for any command group.
```

Comment:
As a TD, I think i might find that confusing.

How about changing this:

```
        - ``group`` - The name for a group this command should be considered a member of.

        - ``group_default`` - Boolean value indicating whether this command should represent the group
          as a whole. Setting this value to True indicates that this is the command to run and display
          in applications that show a single representation for any command group.
```

Into something this:

```
**Grouping commands into collections**

It is possible to group several commands into a collection. Such a collection is called
a *group*.  For example, you may have three separate commands to launch Maya 2017, 
Maya 2016 and Maya 2015, all grouped under a 'Launch Maya' group. It is up to each 
engine to implement this specification in a suitable way but typically, it would displayed as 
a a "Launch Maya" menu with three sub menu items to represent each version of Maya.

Each group has a concept of a group default - this is what would get executed if you click 
on the 'Launch Maya' group.

To register commands with groups, pass the following two parameters:

        - ``group`` - The name for a group this command should be considered a member of.

        - ``group_default`` - Boolean value indicating whether this command should represent the group
          as a whole. 

.. note:: It is up to each engine to implement this in an appropriate way. Some engines 
may not support grouping.

.. note:: If more than one group default is specified, the engine should treat the last registered
group default command the default one. Conversely, if no group default has been set for a group,
the engine should treat the last registered command as the default.

```

For the logic of the group default - i am not sure what makes the most sense, (maybe it's not the last one!) - but whatever logic we want engine developers to implement and app developers to expect engines to implement must be clearly described in these docs.